Source/WebCore/html/TextFieldInputType.cpp

```diff
@@ -911,6 +911,8 @@ void TextFieldInputType::dataListMayHaveChanged()
     if (!element())
         return;
     m_dataListDropdownIndicator->setInlineStyleProperty(CSSPropertyDisplay, element()->list() ? CSSValueBlock : CSSValueNone, true);
+    if (element()->list() && visibleValue().length())
+        displaySuggestions(DataListSuggestionActivationType::DataListMayHaveChanged);
```

Comment:
The `<datalist>` contents could change even when the input is empty, so I'm not sure why the `visibleValue` check is necessary?

Reply:
You are right, checking for the input being focused is better, that is in the new version. The reason I was doing this kind of check is tested by datalist-textinput-dynamically-add-options.html, i.e. don't just show the picker with the new suggestions when the user is not yet interacting with the input only because datalist content was changed.